src/System.Management.Automation/utils/PsUtils.cs

```diff
@@ -90,45 +90,91 @@ internal static ProcessModule GetMainModule(Process targetProcess)
             return mainModule;
         }
 
+        // Cache of the current process' parentId 
+        private static int? s_currentParentProcessId;
+        private static readonly int s_currentProcessId = Process.GetCurrentProcess().Id;
+
         /// <summary>
-        /// Retrieve the parent process of a process.
-        /// 
-        /// Previously this code used WMI, but WMI is causing a CPU spike whenever the query gets called as it results in 
-        /// tzres.dll and tzres.mui.dll being loaded into every process to conver the time information to local format.
-        /// For perf reasons, we result to P/Invoke.
-        /// </summary>
-        ///
-        /// <param name="current">The process we want to find the
-        /// parent of</param>
-        internal static Process GetParentProcess(Process current)
-        {
+        /// Gets the parent process id for the specified process
+        /// </summary>        
+        /// <remarks>
+        /// Caching is used for the current process
+        /// </remarks>
+        /// <param name="processId">the id of the process whose parent id should be retreived</param>
+        /// <returns>0 on failure, else a parent process id. Note that the parent process id may have been recycled</returns>
+        internal static int GetParentProcessId(int processId)
```

Comment:
This is very close to being a duplication of GetParentProcess(). I think it'd be OK to add the caching logic to GetParentProcessIdNative. Also, it doesn't have the start time validation that GetParentProcess has, so we should make sure we don't expose anything that looks like a useful helper function without this validation.